src/Microsoft.Health.Fhir.Api/Features/Routing/KnownRoutes.cs

```diff
@@ -52,6 +52,9 @@ internal class KnownRoutes
         public const string ConvertData = "$convert-data";
         public const string ConvertDataOperationDefinition = OperationDefinition + "/" + OperationsConstants.ConvertData;
 
+        public const string Everything = "$everything";
+        public const string EverythingResourceTypeById = ResourceTypeById + "/" + Everything;
```

Comment:
>EverythingResourceTypeById [](http://example.com/codeflow?start=28&length=26)

I think it's better restict on route level, than catch wrong resource type in controller.

So I would suggest to have 
`"Pateint"+"/"+IdRouteSegment+"/"+Everithing`

Reply:
Agree. I've moved the restrictions on route level in the update.